Get CORS support working with XHR preflight

paulmelnikow opened the pull request:
Without this patch, http-server returns 404 on the preflight request (method: 'OPTIONS'). You can see the problem by loading an XHR request in Chrome.

paulmelnikow commented:
ping @indexzero, can you get this merged?

indexzero commented:
@paulmelnikow can you rebase off `master`?

paulmelnikow commented:
done

indexzero commented:
@paulmelnikow read a little closer, there is actually a `--cors` option in the CLI, but in this implementation the `corser` middleware is added by default. Can you make it respond to the option? Then I'll merge. 

Bonus points if you write tests.

paulmelnikow commented:
Ah, makes sense. I'll take another pass.

paulmelnikow commented:
Updated, now honoring the cors option. Includes a test which fails if you remove `cors: true`. Moved the cors header config into http-server.js so it works in the test.

indexzero commented:
Awesome! Merging. Thanks for writing the tests :100:
